### PR TITLE
feat: load chapter from url for creating new event from chapter

### PR DESCRIPTION
**Pull request by gikf**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `main` branch of Chapter.

---

- Events can be added from chapter dashboard, based on the url - `dashboard/chapters/${id}/new_event`.
- Otherwise when using `dashboard/events/new`, `chapter_id` defaults to `1`, the same as currently.
- For clarity chapter name is displayed above the form.

### CODE REVIEW

File: client/src/pages/dashboard/chapters/[id]/new_event.tsx [hunk 1]
```diff
@@ -0,0 +1,2 @@
+import { NewEventPage } from '../../../../modules/dashboard/Events';
+export default NewEventPage;
```
Comment (ojeytonwilliams) [inline client/src/pages/dashboard/chapters/[id]/new_event.tsx:2]: With this `dashboard/events/new` and `dashboard/chapters/1/new_event` are equivalent. I'm not sure having two urls that're essentially the same is the way to go.

What do you think to getting rid of `dashboard/events/new`?

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Hey @gikf thanks for this, it's looking great.  Thoughts below:

File: client/src/modules/dashboard/Events/components/EventForm.tsx [hunk 2]
```diff
@@ -34,7 +36,15 @@ import {
 } from './EventFormUtils';
 
 const EventForm: React.FC<EventFormProps> = (props) => {
-  const { onSubmit, data, loading, submitText } = props;
+  const { onSubmit, data, loading, submitText, chapterId } = props;
+  const chapter_id = chapterId ?? data?.chapter_id ?? 1;
```
[the remark this answers is not in the document]
Comment (gikf) [reply client/src/modules/dashboard/Events/components/EventForm.tsx]: This was to conform three cases - new event with `chapterId` passed, editing event, new event without chapter.

With requiring `chapterId` prop, there's now this:
https://github.com/freeCodeCamp/chapter/blob/4a581c09f5c9e7b7d6cef5c7e6926684b4bb5bcf/client/src/modules/dashboard/Events/pages/EditEventPage.tsx#L82-L94
Removing the need for `chapter_id` in `data` is surprisingly troublesome, unless `chapter_id` is made optional in `IEventData` type. I didn't have much success with finding easy way for removing `chapter_id` from `IEventData` completely.

File: client/src/pages/dashboard/chapters/[id]/new_event.tsx (hunk 1 above)
Comment (gikf) [reply client/src/pages/dashboard/chapters/[id]/new_event.tsx:2]: Right, `dashboard/events/new` with some _arbitrary_ default chapter is a bit obsolete, when chapter can be picked up from url.

File: client/src/modules/dashboard/Events/components/EventForm.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (ojeytonwilliams) [reply client/src/modules/dashboard/Events/components/EventForm.tsx]: ...actually, in `EventForm` we could pass `(data) => onSubmit(data, chapterId)` as the callback for `handleSubmit` (on line 113).  That means we'd no longer be using a slightly hacky approach (default value for a non-existent field) and we could stop including `chapter_id` in `data`.

Does that make any kind of sense or am I just off in the weeds?

[the remark this answers is not in the document]
Comment (gikf) [reply client/src/modules/dashboard/Events/components/EventForm.tsx]: I think I got it, I've pushed commit with this change.

File: client/src/modules/dashboard/Events/components/EventFormUtils.ts [hunk 3]
```diff
@@ -132,7 +132,8 @@ export interface EventFormData {
 
 export type IEventData = Pick<
   Event,
-  keyof Omit<EventFormData, 'venue_id' | 'tags' | 'sponsors'> | 'id'
+  | keyof Omit<EventFormData, 'venue_id' | 'tags' | 'sponsors' | 'chapter_id'>
+  | 'id'
```
Comment (ojeytonwilliams) [inline client/src/modules/dashboard/Events/components/EventFormUtils.ts:136]:
```suggestion
  keyof Omit<EventFormData, 'venue_id' | 'tags' | 'sponsors'> | 'id'
```
Nit: `chapter_id` isn't in `EventFormData`

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Tiniest of nitpicks. Rhis is looking great

**Review by ojeytonwilliams — APPROVED** [review]

**Comment by ojeytonwilliams** [comment]
Perfect, thanks!

**Pull request merged**